index.bs

```diff
@@ -1728,23 +1639,19 @@ partial interface MLGraphBuilder {
 The {{MLGraph}} interface represents a compiled computational graph. A compiled graph once constructed is immutable and cannot be subsequently changed.
 
 <script type=idl>
-dictionary MLInput {
-  required (MLBufferView or WebGLTexture or GPUTexture) data;
-  sequence<long> dimensions;
-};
+typedef (MLBufferView or WebGLTexture or GPUTexture) MLResource;
 
-dictionary MLOutput {
-  (MLBufferView or WebGLTexture or GPUTexture) data;
+dictionary MLInput {
+  required MLResource resource;
   sequence<long> dimensions;
 };
 
-typedef record<DOMString, MLInput> MLNamedInputs;
-typedef record<DOMString, MLOutput> MLNamedOutputs;
+typedef record<DOMString, (MLResource or MLInput)> MLNamedInputs;
```

Comment:
I think this is a good idea as it removes a level of indirection in the case when just a resource is passed in, but should the `dimensions` field in `MLInput` now be required?

Reply:
Good catch, it should be required. I fixed it in the new commit. Please take a look.